src/main/java/com/merck/rdf2x/jobs/stats/StatsJob.java

```diff
@@ -80,12 +95,45 @@ public void run() {
             counts.sortByKey(false).take(100).forEach(uriCount -> {
                 log.info(uriCount.toString());
             });
+            subjectURICount = counts.count();
+            stats.remove(SUBJECT_URI_COUNT);
+        }
+
+        if(stats.contains(PREDICATE_URI_COUNT)) {
             log.info("----------------------------");
-            log.info("Total Distinct Subject URIs: {}", counts.count());
+            log.info("Predicate URI stats:");
+            JavaPairRDD<Long, String> counts = QuadCounter.countByPredicateURI(quads).mapToPair(Tuple2::swap);
+            counts.sortByKey(false).take(100).forEach(uriCount -> {
+                log.info(uriCount.toString());
+            });
+            predicateURICount = counts.count();
+            stats.remove(PREDICATE_URI_COUNT);
+        }
+
+        if(stats.contains(OBJECT_URI_COUNT)) {
             log.info("----------------------------");
-            stats.remove(SUBJECT_URI_COUNT);
+            log.info("Object URI stats:");
+            JavaPairRDD<Long, String> counts = QuadCounter.getObjectURI(quads).mapToPair(Tuple2::swap);
+            counts.sortByKey(false).take(100).forEach( uriCount -> {
+                log.info(uriCount.toString());
+            });
+            objectURICount = counts.count();
+            stats.remove(OBJECT_URI_COUNT);
         }
 
+        printStats();
+
+    }
+
+    /**
+     * Prints stats for distinct counts of {@link Quad} properties i.e. subject, predicate and object
+     */
+    private void printStats() {
+        log.info("---------------------------------");
+        log.info("Total Distinct Subject URIs: {}", subjectURICount);
```

Comment:
Can you print the statistics only for the requested fields? You could set them to `null` and only print it if not `null`. It is a bit confusing to have them set to `0` if they are not requested.